cmd/ak/cmd/manifest/apply.go

```diff
@@ -2,100 +2,45 @@ package manifest
 
 import (
 	"fmt"
-	"io"
 	"os"
-	"strings"
 
 	"github.com/spf13/cobra"
-	"github.com/xeipuuv/gojsonschema"
-	"gopkg.in/yaml.v3"
 
-	"go.autokitteh.dev/autokitteh/backend/apply"
 	"go.autokitteh.dev/autokitteh/cmd/ak/common"
-	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkmanifest"
 )
 
-type output struct {
-	Logs       any `json:"logs"`
-	Operations any `json:"operations"`
-}
-
-var dryRun, noValidate bool
+var dryRun bool
 
 var applyCmd = common.StandardCommand(&cobra.Command{
-	Use:     "apply [file] [--no-validate] [--dry-run]",
+	Use:     "apply [file] [--no-validate] [--from-scratch] [--dry-run]",
 	Short:   "Apply project configuration from file or stdin",
 	Aliases: []string{"a"},
 	Args:    cobra.MaximumNArgs(1),
 
 	RunE: func(cmd *cobra.Command, args []string) error {
-		var (
-			path string
-			data []byte
-			err  error
-		)
-		if len(args) == 0 {
-			data, err = io.ReadAll(os.Stdin)
-			if err != nil {
-				return fmt.Errorf("stdin: %w", err)
-			}
-		} else {
-			path = args[0]
-			data, err = os.ReadFile(path)
-			if err != nil {
-				return err
-			}
-		}
-
-		var root apply.Root
-		if err := yaml.Unmarshal(data, &root); err != nil {
-			return fmt.Errorf("invalid YAML input: %w", err)
+		data, path, err := common.Consume(args)
+		if err != nil {
+			return err
 		}
 
-		if !noValidate {
-			res, err := gojsonschema.Validate(
-				gojsonschema.NewStringLoader(apply.JSONSchemaString),
-				gojsonschema.NewGoLoader(&root),
-			)
-			if err != nil {
-				return fmt.Errorf("validate: %w", err)
-			}
-			if !res.Valid() {
-				msg := strings.Join(kittehs.Transform(res.Errors(), err2str), "\n")
-				return fmt.Errorf("invalid YAML semantics: %s", msg)
-			}
+		actions, err := plan(data, path)
```

Comment:
If this function is called only here, why is it implemented in a different file?

Reply:
oops - now used in plan cmd too.